src/app/components/task-button/task-button.component.ts

```diff
@@ -62,11 +64,30 @@ export class TaskButtonComponent implements OnInit {
       { display: 'inline-block' }
     );
   }
+
   onTaskButtonClick({ dataJob, id }: TaskEventData): void {
+
     switch (dataJob) {
+      case this.constantsService.COMPLETE:    
+        this.taskButtonClick.emit({id, dataJob});
+        break;
+  
       case this.constantsService.DELETE_TODO:
-        this.taskButtonClick.emit({ id, dataJob });
+        this.taskButtonClick.emit({id, dataJob});
         break;
     }
   }
+
+  visibilityHandle({dataJob}: {dataJob: any}, task: Task): boolean {
+    switch (dataJob) {
+      case 'edit':
```

Comment:
Also it is a good practice to use string from constant instead directly using them.

Reply:
done